rust_src/src/process.rs

```diff
@@ -320,4 +325,20 @@ pub fn set_process_query_on_exit_flag(process: LispObject, flag: LispObject) ->
     flag
 }
 
+/// Return the exit status of PROCESS or the signal number that killed it.
+/// If PROCESS has not yet exited or died, return 0.
+#[lisp_fn]
+pub fn process_exit_status(process: LispObject) -> LispObject {
+    let p_ref = process.as_process_or_error();
+    if p_ref.raw_status_new() {
+        unsafe { update_status(p_ref.as_ptr()) };
+    }
+    let status = LispObject::from(p_ref.status);
+    if let Some(s) = status.as_cons() {
+        s.cdr().as_cons().unwrap().car()
```

Comment:
That `unwrap` will not have the same behaviour as the `eassert(CONSP)` from the C code.

Reply:
I just realized that there is a macro 
`# define CONSP(x) lisp_h_CONSP (x)` 
and a function 
```
INLINE bool
(CONSP) (Lisp_Object x)
{
  return lisp_h_CONSP (x);
}
```

Now I'm consued :D
I also can't find `eassert(CONSP)`.

Reply:
```
#define lisp_h_XCAR(c) XCONS (c)->car
#define lisp_h_XCDR(c) XCONS (c)->u.cdr
#define lisp_h_XCONS(a) \
   (eassert (CONSP (a)), (struct Lisp_Cons *) XUNTAG (a, Lisp_Cons))
```

Remember there isa space before the parens when grepping.

Reply:
My bad. I only looked at `CONSP` and not at `XCAR` and `XCDR`....sorry.

lisp.h is really confusing...

Reply:
yes it is! No worries.